modules/carto/src/api/fetch-map.ts

```diff
@@ -80,19 +80,24 @@ async function _fetchMapDataset(
       if (type === 'table') {
         dataset.data = await vectorTableSource({...options, columns, tableName: source});
       } else if (type === 'query') {
-        dataset.data = await vectorQuerySource({...options, sqlQuery: source, queryParameters});
+        dataset.data = await vectorQuerySource({
+          ...options,
+          columns,
+          sqlQuery: source,
+          queryParameters
+        });
       }
     } else if (spatialDataType === 'h3') {
       const options = {...globalOptions, aggregationExp, aggregationResLevel, spatialDataColumn};
       if (type === 'table') {
-        dataset.data = await h3TableSource({...options, columns, tableName: source});
+        dataset.data = await h3TableSource({...options, tableName: source});
```

Comment:
i've removed it accidentally, but now i'm not sure if it makes sense to have `columns` for spatial-indexes when properties have to be specified explicitly with `aggregationExp`

Reply:
As you say, `columns` makes no sense for spatial-indexes and the Maps API doesn't take it into account